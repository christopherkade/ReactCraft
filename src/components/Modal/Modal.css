.rc-modal {
  display: flex;
  flex-direction: column;
  background-color: rgb(22, 17, 14, 0.6);
  text-align: center;
  padding: 0.5rem 4rem;
  box-shadow: inset 1px 1px 7rem 0rem black;
  border: 0.25rem ridge #8C8C8C;
  width: 500px;
}

.rc-modal-button-container {
  display: flex;
  flex-wrap: wrap;
  justify-content: space-evenly;
  margin: 1rem 0 1rem 0;
}

.rc-modal-button-container > button {
  padding-left: 5rem;
  padding-right: 5rem;
}
