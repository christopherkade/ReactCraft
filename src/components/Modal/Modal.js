import React from "react"
import PropTypes from 'prop-types'
import styled from "styled-components"

import { Title } from "../Title"
import { Button } from "../Button"

const Wrapper = styled.div`
  position: relative;
  display: flex;
  flex-direction: column;
  background-color: rgb(22, 17, 14, 0.6);
  text-align: center;
  padding: 0.5rem 4rem;
  box-shadow: inset 1px 1px 7rem 0rem black;
  border: 0.25rem ridge #8C8C8C;
  width: 500px;
`

const ButtonWrapper = styled.div`
  display: flex;
  flex-wrap: wrap;
  justify-content: space-evenly;
  margin: 1rem 0 1rem 0;

  & > button {
    padding-left: 5rem;
    padding-right: 5rem;
  }
`

const TopLeft = styled.div`
  position: absolute;
  top: 0;
  left: 0;
  width: 0.25rem;
  height: 0.25rem;
  background-color: #8C8C8C;
  z-index: 100;
  transform: rotate(45deg) translate(-10px);
  border: 2px ridge #8C8C8C;
`

const TopRight = styled.div`
  position: absolute;
  top: 0;
  right: 0;
  width: 0.25rem;
  height: 0.25rem;
  background-color: #8C8C8C;
  z-index: 100;
  transform: rotate(45deg) translateY(-10px);
  border: 2px ridge #8C8C8C;
`

const BottomRight = styled.div`
  position: absolute;
  bottom: 0;
  right: 0;
  width: 0.25rem;
  height: 0.25rem;
  background-color: #8C8C8C;
  z-index: 100;
  transform: rotate(45deg) translate(10px);
  border: 2px ridge #8C8C8C;
`

const BottomLeft = styled.div`
  position: absolute;
  bottom: 0;
  left: 0;
  width: 0.25rem;
  height: 0.25rem;
  background-color: #8C8C8C;
  z-index: 100;
  transform: rotate(45deg) translateY(10px);
  border: 2px ridge #8C8C8C;
`

const Modal = ({
  title,
  okText,
  cancelText,
  onOk,
  onCancel
}) => {
  return (
    <Wrapper>
      <Title>{title}</Title>

      <ButtonWrapper>
        {okText ? <Button onClick={onOk}>{okText}</Button> : null}
        {cancelText ? <Button onClick={onCancel}>{cancelText}</Button> : null}
      </ButtonWrapper>

      <TopLeft />
      <TopRight />
      <BottomRight />
      <BottomLeft />
    </Wrapper>
  )
}

Modal.propTypes = {
  title: PropTypes.string.isRequired,
  okText: PropTypes.string,
  onOk: PropTypes.func,
  cancelText: PropTypes.string,
  onCancel: PropTypes.func
};

export default Modal
