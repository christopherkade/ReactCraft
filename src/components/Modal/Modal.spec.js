import React from 'react';
import Modal from './Modal';
import { render, fireEvent } from 'react-testing-library';

describe('<Modal />', () => {
  describe('Snapshots', () => {
    it('should match snapshot', () => {
      const { container } = render(<Modal title="My awesome title"></Modal>);
      expect(container).toMatchSnapshot();
    });
  });

  describe('tile prop', () => {
    it('should display Logging in as the modal\'s title', () => {
      const { container } = render(<Modal title="Logging in" />);
      expect(container.querySelector('h1').textContent).toBe('Logging in');
    });
  });

  describe('onClick prop', () => {
    it('should call onOk function when the Ok button is clicked', () => {
      const onClickMock = jest.fn();
      const { getByText } = render(<Modal onOk={onClickMock} okText="Ok" title="onClick test" />);
      fireEvent.click(getByText('Ok'));
      expect(onClickMock).toHaveBeenCalled();
    });

    it('should call onCancel function when the Cancel button is clicked', () => {
      const onClickMock = jest.fn();
      const { getByText } = render(<Modal onCancel={onClickMock} cancelText="Cancel" title="onClick test" />);
      fireEvent.click(getByText('Cancel'));
      expect(onClickMock).toHaveBeenCalled();
    });
  });
});
