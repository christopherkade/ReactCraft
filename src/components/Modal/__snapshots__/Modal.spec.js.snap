// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Modal /> Snapshots should match snapshot 1`] = `
<div>
  <div
    class="sc-htpNat fyMTWp"
  >
    <h1
      class="sc-bdVaJa cUfhWq"
    >
      My awesome title
    </h1>
    <div
      class="sc-bxivhb bzVgvM"
    />
    <div
      class="sc-ifAKCX cSZYAr"
    />
    <div
      class="sc-EHOje fZDMET"
    />
    <div
      class="sc-bZQynM cTONNa"
    />
    <div
      class="sc-gzVnrw hHuWZU"
    />
  </div>
</div>
`;
